### PR TITLE
Harmonize naming for abstract classes

### PR DESCRIPTION
**Pull request by mweisgut**
We already use the names `AbstractOperator`, `AbstractIndex`, `AbstractLQPNode` and others. The purpose of this PR is to meet the naming conventions for the following classes which have multiple differing implementations (except `AbstractSegmentType`):
- `BaseEncodedSegment` -> `AbstractEncodedSegment`
- `BasePointAccessSegmentIterator` -> `AbstractPointAccessSegmentIterator`
- `BaseSegment` -> `AbstractSegment`
- `BaseSegmentIterator` -> `AbstractSegmentIterator`
- `BaseSegmentType` -> `AbstractSegmentType`

Related convention:
>Classes that are used only to have a non-templated base class are named BaseXY (e.g., BaseColumn), while classes that have multiple differing implementations are named AbstractXY (e.g., AbstractOperator)

### CODE REVIEW

File: src/lib/storage/abstract_segment.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (mrks) [inline src/lib/storage/abstract_segment.cpp:1]: remove empty line

**Comment by mrks** [comment]
Where did we have a `BaseStatisticsObject`?

**Comment by mweisgut** [comment]
> Where did we have a `BaseStatisticsObject`?

Good point. We didn't use it. I haven't checked `AbstractStatisticsObject` thoroughly while grepping the diff. I will update the description.

**Comment by mweisgut** [comment]
Related software engineering SE question/discussion: https://softwareengineering.stackexchange.com/questions/347944/naming-procedure-conventions-where-to-place-adjective-in-a-verb-followed-by-an

**Pull request merged**